lib/https.js

```diff
@@ -93,11 +93,13 @@ function createServer(opts, requestListener) {
 
 function createConnection(port, host, options) {
   if (port !== null && typeof port === 'object') {
-    options = port;
+    options = { ...port };
```

Comment:
It does not harm but there is no need to copy here. `createConnection(options, host)` is not a documented/supported signature and if `port` is already an object tls.connect(port) does not mutate it.

Reply:
Nice point. I will fix that now. Generally, do we ignore not documented signatures or should we take care of them?